openbb_platform/providers/government_us/openbb_government_us/models/treasury_auctions.py

```diff
@@ -0,0 +1,94 @@
+"""US Government Treasury Auctions."""
+
+from datetime import (
+    datetime,
+    timedelta,
+)
+from typing import Any, Dict, List, Optional
+
+import pandas as pd
+import requests
+from openbb_provider.abstract.fetcher import Fetcher
+from openbb_provider.standard_models.treasury_auctions import (
+    USTreasuryAuctionsData,
+    USTreasuryAuctionsQueryParams,
+)
+from openbb_provider.utils.helpers import get_querystring
+
+
+class GovernmentUSTreasuryAuctionsQueryParams(USTreasuryAuctionsQueryParams):
+    """
+    US Treasury Auctions Query Params
+
+    Source: https://www.treasurydirect.gov/
+    """
+
+    __alias_dict__ = {
+        "start_date": "startDate",
+        "end_date": "endDate",
+        "security_type": "type",
+        "page_size": "pagesize",
+        "page_num": "pagenum",
+    }
+
+
+class GovernementUSTreasuryAuctionsData(USTreasuryAuctionsData):
+    """US Treasury Auctions Data."""
+
+
+class GovernmentUSTreasuryAuctionsFetcher(
+    Fetcher[
+        GovernmentUSTreasuryAuctionsQueryParams,
+        List[GovernementUSTreasuryAuctionsData],
+    ]
+):
+    """US Treasury Auctions Fetcher."""
+
+    @staticmethod
+    def transform_query(
+        params: Dict[str, Any]
+    ) -> GovernmentUSTreasuryAuctionsQueryParams:
+        """Transform query params."""
+        return GovernmentUSTreasuryAuctionsQueryParams(**params)
+
+    @staticmethod
+    def extract_data(
+        query: GovernmentUSTreasuryAuctionsQueryParams,
+        credentials: Optional[Dict[str, str]],
+        **kwargs: Any,
+    ) -> List[Dict]:
+        """Extract the raw data from Treasury Direct API."""
+
+        base_url = "https://www.treasurydirect.gov/TA_WS/securities/search?"
+
+        if not query.start_date:
+            query.start_date = datetime.now() - timedelta(days=90)
+        if not query.end_date:
+            query.end_date = datetime.now()
+
+        _query = query.model_dump()
+        _query["startDate"] = (
+            _query["startDate"].strftime("%m/%d/%Y") if query.start_date else None
+        )
+        _query["endDate"] = (
+            _query["endDate"].strftime("%m/%d/%Y") if _query["endDate"] else None
+        )
+        query_string = get_querystring(_query, [])
```

Comment:
Can you handle this Query part in the transform_query step please?

Reply:
Actually, if I do this step in the `transform_query` it simply returns the date format to the original formatting, YYYY-MM-DD.

The Output, for the URL, needs to be formatted specifically like that.

Reply:
What about the start_date and end_date logic?

Reply:
I can move the part that sets the default start/end.